treescript/src/treescript/versionmanip.py

```diff
@@ -68,6 +68,27 @@ def get_version(file_, parent_directory=None):
 async def bump_version(config, task, repo_path):
     """Perform a version bump.
 
+    This function takes its inputs from task by using the ``get_version_bump_info``
+    function from treescript.task. Using `next_version` and `files`, then
+    calls do_version_bump to perform the work.
+
+    Args:
+        config (dict): the running config
+        task (dict): the running task
+        repo_path (str): the source directory
+
+    Returns:
+        int: the number of commits created.
+
+    """
+    bump_info = get_version_bump_info(task)
+
+    return await do_bump_version(config, repo_path, bump_info["files"], bump_info["next_version"], get_dontbuild(task), commit=True)
+
+
+async def do_bump_version(config, repo_path, files, next_version, dontbuild, commit=True):
```

Comment:
We'd then need to duplicate those commits inside `bump_version` and twice in `apply_rebranding` - the contents of the old `bump_version` were drawn out for re-use so that it could be called from the task payload and also in a merge.

Reply:
The call in `apply_rebranding` always passes `commit=False` though. I'm suggesting moving the last `if changed and commit` block into the one caller that currently passes `commit=True`. I don't think there needs to be any duplication.